### ISSUE TITLE
Segfault on MAC OSX 10.8 with BPMC BF1

### ISSUE THREAD

**Issue by TrooperT** [luke-jr/bfgminer#300]
![screen shot 2013-10-10 at 12 32 14 pm](https://f.cloud.github.com/assets/1194617/1308183/0c01248e-31ca-11e3-977b-8e7917ee3d60.png)
compiled from git source using nwools brew script with --HEAD.

command run:
bfgminer --config ~/bfgminer.conf -S bigpic:/dev/cu.usbmodem1d11431 -D 2>debug.log

log available at: https://gist.github.com/Taugeran/6921416

**Comment by luke-jr** [comment]
Can you get a backtrace?

**Comment by TrooperT** [comment]
if i knew how yes. im going to try it by itself with no BES attached b/c i can get it to run on a linux box by itself with the same code

**Comment by TrooperT** [comment]
update: still does it while the only mining device on the system macOSX

**Comment by luke-jr** [comment]
```
gdb --args <bfgminer command>
run
(wait for it to crash)
bt
thr app all bt
```

**Comment by TrooperT** [comment]
https://gist.github.com/Taugeran/6923335

login to thr app all bt

**Comment by luke-jr** [comment]
Seems to be missing quite a bit, including the crash info itself? :/

**Comment by TrooperT** [comment]
thats all that it printed to screen following your instructions.

if it doesnt work on mac im not bothered. ive got an rPI just sitting around that can run it ^_^

**Comment by luke-jr** [comment]
I don't even see where you typed the 'thr app all bt' command, so something went wrong making the log I think :/

**Comment by TrooperT** [comment]
Ill redo it after work

**Comment by luke-jr** [comment]
Maybe with --text-only will work better

**Comment by TrooperT** [comment]
Bfgminer <args> --text-only

That?

Sam Johnson
P: 937.825.6346
E: Samuel.Johnson.Bungie@Gmail.com

**Comment by luke-jr** [comment]
Right, with the same gdb setup as before.

**Comment by TrooperT** [comment]
http://imgur.com/a/Kp8YF

album of screens after each command looks about the same

**Comment by luke-jr** [comment]
Is it possible to rebuild with debug symbols, and meet me on IRC to interactively debug it?

**Comment by luke-jr** [comment]
@Taugeran I have a unit to test with myself, and am unable to reproduce (on Linux). Can you try with debug symbols? :/

**Comment by TrooperT** [comment]
that would be

CFLAGS="-g -O3" ./configure ..... 

?

**Comment by luke-jr** [comment]
./configure ... CFLAGS='-O0 -ggdb'

**Comment by TrooperT** [comment]
brewed with the command: brew install -vd --env=std bfgminer --HEAD

got more from the app all bt this time:

https://gist.github.com/Taugeran/7148480

**Comment by nwoolls** [comment]
You can download the Homebrew script for bfgminer here: https://raw.github.com/nwoolls/homebrew-xgminer/master/bfgminer.rb

Just save that file locally. Then you can edit it with any text editor, e.g. TextEdit. There is a block in there that includes the ./configure arguments. Add the one that Luke is suggesting and save the changes.

After that you can install your local copy of the Homebrew formula by doing:

``` bash
brew install ~/path/to/bfgminer.rb
```

**Comment by TrooperT** [comment]
thats actually how ive been compiling it. 

in the install block I've added:

ENV['CFLAGS']="-ggdb -O0"

but without --env=std while brew installing, it strips those flags :/... as far as i can tell from the verbose output

**Comment by luke-jr** [comment]
@Taugeran Even with your --env=std, you've failed to produce results with debugging symbols :(

**Comment by TrooperT** [comment]
alas apple has failed me :/ no worries. if its ok ill close this since im sure only a microscopic fraction of people mine with asics on a mac. if they have asics they're probably going for efficiency ( hint: not a mac).

and besides ive got em running on an rpi 

but if its anything to you i have something in the works for you :)

**Comment by luke-jr** [comment]
If you get time, I'd still like to fix this some day :)

**Comment by TrooperT** [comment]
will keep it in mind :)

**Comment by nwoolls** [comment]
If you are unable to get Homebrew to pass in the proper flags you could also try using the shell scripts here to build bfgminer with those flags:

https://github.com/nwoolls/xgminer-osx

There are instructions. You'll need to edit the build_miner.sh script to add set the CFLAGS argument.

**Comment by TrooperT** [comment]
cool deal. will give her a try. and if all else. its mac and asics. not a smart plan for power efficiency anyway (85W laptop yada yada ...)

**Comment by TrooperT** [comment]
we found it. we found it!!! thank you nwoolls for the tip on xgminer-osx. it was the saviours grace. the whole problem was 3 lines of timing code in driver-bigpic.c :S

**Comment by nwoolls** [comment]
Glad to hear it @Taugeran

**Comment by TrooperT** [comment]
edit: those 3 lines shouldn't have even been there regardless of OS. ljr was surprised this wasn't showing up on windows or generic linux

**Comment by TrooperT** [comment]
closing issue :)

**Comment by luke-jr** [comment]
For reference, fixed in efe47d5
